server/channels/app/integration_action.go

```diff
@@ -460,6 +460,10 @@ func (a *App) OpenInteractiveDialog(request model.OpenDialogRequest) *model.AppE
 		return appErr
 	}
 
+	if dialogErr := request.IsValid(); dialogErr != nil {
+		return model.NewAppError("OpenInteractiveDialog", "app.open_interactive_dialog.invalid_request", nil, "", http.StatusBadRequest).Wrap(dialogErr)
+	}
```

Comment:
After giving this another thought, I wonder if the code should be more liberal and not error out. In case of bugs in `IsValid` this currently blocks integration developers from open a dialog that the server deems faulty. 

I'm leaning towards logging a warning message instead. @annaos @mickmister what do you think?

Reply:
This is a good idea. Errors such as exceeding the maximum length or having the wrong default value are not as critical and should not block the opening of a dialog. However, errors like the wrong element type or an empty URL are critical and will result in a broken dialog, correct?

Should I still return an error for important cases (tbd) and log a warning for the rest? @hanzei @mickmister

Reply:
Good question. I wonder if blocking the dialogue creation at all is helpful. It is not a great UX (as the user doesn't understand why the dialogue won't open and can't fix it).

So I'm leaning towards never blocking the request.

Reply:
I logged an error level message. I hope that's okay with you. I couldn't find a way to check a log message in the test. If you know of one, please let me know. Otherwise everything done.